Discourage `document.writeln()` like `document.write()`

caugner opened the pull request:
Discourage `document.writeln()` for the same reasons as `document.write()`

See: https://github.com/whatwg/html/issues/7977#issuecomment-2979225203

- [ ] At least two implementers are interested (and none opposed):
   * …
   * …
- [ ] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * … <!-- If these tests are tentative, link a PR to make them non-tentative. -->
- [ ] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chromium: …
   * Gecko: …
   * WebKit: …
   * Deno (only for timers, structured clone, base64 utils, channel messaging, module resolution, web workers, and web storage): …
   * Node.js (only for timers, structured clone, base64 utils, channel messaging, and module resolution): …
- [ ] Corresponding [HTML AAM](https://w3c.github.io/html-aam/) & [ARIA in HTML](https://w3c.github.io/html-aria/) issues & PRs:
- [ ] [MDN issue](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) is filed: …
- [ ] The top of this comment includes a [clear commit message](https://github.com/whatwg/meta/blob/main/COMMITTING.md) to use. <!-- If you created this PR from a single commit, Github copied its message. Otherwise, you need to add a commit message yourself. -->

(See [WHATWG Working Mode: Changes](https://whatwg.org/working-mode#changes) for more details.)


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://whatpr.org/html/11384/dynamic-markup-insertion.html" title="Last updated on Jun 19, 2025, 9:52 PM UTC (f6a112b)">/dynamic-markup-insertion.html</a>  ( <a href="https://whatpr.org/html/11384/41a3b79...f6a112b/dynamic-markup-insertion.html" title="Last updated on Jun 19, 2025, 9:52 PM UTC (f6a112b)">diff</a> )

annevk commented on source:
```diff
@@ -116048,6 +116048,9 @@ document.body.appendChild(frame)</code></pre>
     character. If necessary, calls the <code data-x="dom-document-open">open()</code> method
     implicitly first.</p>
 
+    <p class="warning">This method has very idiosyncratic behavior. <strong>Use of this method is
+    strongly discouraged, for the same reasons as <code data-x="dom-document-write">write()</code>.</strong></p>
```
```suggestion
    <p class="warning">This method has very idiosyncratic behavior. <strong>Use of this
    method is strongly discouraged, for the same reasons as <code
    data-x="dom-document-write">document.write()</code>.</strong></p>
```

annevk commented:
@zcorpan could you please work with @caugner on getting the Participation check resolved?

annevk commented:
You need to make your membership of the organization public. Also please don't create merge commits in the future. Especially when they're not needed.

caugner commented:
> You need to make your membership of the organization public.

Done, I was not aware.

> Also please don't create merge commits in the future. Especially when they're not needed.

Sure, no problem. My intention here was to re-trigger the checks (which didn't work). I have rebased and squashed the two commits now.

annevk commented:
Thanks for fixing this!
